adapters/repos/db/lsmkv/commitlogger_parser_collection.go

```diff
@@ -12,60 +12,85 @@
 package lsmkv
 
 import (
-	"bufio"
 	"encoding/binary"
+	"fmt"
 	"io"
-	"os"
 
 	"github.com/pkg/errors"
-	"github.com/weaviate/weaviate/entities/diskio"
 )
 
 func (p *commitloggerParser) doCollection() error {
-	f, err := os.Open(p.path)
-	if err != nil {
-		return err
-	}
-
-	metered := diskio.NewMeteredReader(f, p.metrics.TrackStartupReadWALDiskIO)
-	p.reader = bufio.NewReaderSize(metered, 1*1024*1024)
-
 	for {
-		var commitType CommitType
+		var version uint8
 
-		err := binary.Read(p.reader, binary.LittleEndian, &commitType)
-		if err == io.EOF {
+		err := binary.Read(p.checksumReader, binary.LittleEndian, &version)
+		if errors.Is(err, io.EOF) {
 			break
 		}
-
 		if err != nil {
-			f.Close()
-			return errors.Wrap(err, "read commit type")
+			return errors.Wrap(err, "read commit version")
 		}
 
-		if CommitTypeCollection.Is(commitType) {
-			if err := p.parseCollectionNode(); err != nil {
-				f.Close()
-				return errors.Wrap(err, "read collection node")
+		switch version {
```

Comment:
what happens if an old version of the code reads an V1 segment?

Reply:
I expect it to fail with an unsupported commitType error. Also it should be the case of an older version trying to read a wal file, and those files are automatically deleted once the segment is successfully created.

Reply:
Not sure on how common downgrades are, but we'll need to explicitly mention that this is a backwards incompatible change.